### PR TITLE
Make and LD stuff

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: config.mk [hunk 1]
```diff
@@ -19,9 +19,11 @@ LDSCRIPT := ldscript
 NON_MATCHING ?= 0
 ifeq ($(PLATFORM),gba)
 ifeq ($(DEBUG),1)
+    PORTABLE     := 0
     NON_MATCHING := 1
     LDSCRIPT := $(LDSCRIPT)_modern.txt
```
Comment (freshollie) [inline config.mk:24]: So why does debug use the modern ldscript?

File: sound/MPlayDef.s [hunk 2]
```diff
@@ -1,430 +1,416 @@
-	.equ	W00,	0x80		@ WAIT
```
Comment (freshollie) [inline sound/MPlayDef.s:1]: Not sure I'm a fan of removing these comments? Can't we just do `@;`?

File: config.mk (hunk 1 above)
Comment (JaceCear) [reply config.mk:24]: I was using that for testing, but you're right that it should be using the same for both.

File: sound/MPlayDef.s (hunk 2 above)
Comment (JaceCear) [reply sound/MPlayDef.s:1]: No we can't, because that'd still be a syntax error for those platforms.

File: Makefile [hunk 3]
```diff
@@ -85,30 +131,46 @@ NODEP ?= 1
...
+
+# no-op
+ASM_PSEUDO_OP_CONV := sed -n 'p'
+else
+# Convert .2byte -> .short and .4byte -> .int
+#  Note that on 32bit architectures .4byte / .int is enough for storing pointers,
+#  but on 64bit targets it would be .8byte / .quad
+#
+# sed expression script by Kurausukun
+ASM_PSEUDO_OP_CONV := sed -i'' -e 's/\.4byte/\.int/g;s/\.2byte/\.short/g'
```
Comment (freshollie) [inline Makefile:151]: Still not confirmed if the `sed -i'' -e` works on linux, it's just something I read on stackoverflow that it should work for both

**Review by freshollie — APPROVED** [review]

**Pull request merged**